crates/runtime/src/memory.rs

```diff
@@ -137,22 +137,25 @@ impl RuntimeLinearMemory for MmapMemory {
         self.maximum
     }
 
-    fn grow_to(&mut self, new_size: usize) -> Option<()> {
+    fn grow_to(&mut self, new_size: usize) -> Result<()> {
+        if new_size == usize::MAX {
```

Comment:
I'd ideally prefer to not have a one-off check for something like this and have it fall out of other checks if possible. The risk I'm worried about is that if this is `usize::MAX` then it flows into `Mmap::accessible_reserved` which goes straight to the OS and will return an error about this being non-page-aligned. I think it'd be best to either have `usize::MAX` here truncated to the OS page alignment (which is still an un-satisfiable request) or something like that.